lib/prog.ml

```diff
@@ -852,23 +854,23 @@ end = struct
 end
 
 and Value : sig
-  type buf = Cstruct_sexp.t [@@deriving sexp]
+  type buf = Cstruct_sexp.t [@@deriving sexp,show]
   (* let buf_to_yojson b = failwith "unimplemented"
   let buf_of_yojson j = failwith "unimplemented" *)
 
   type pkt = {
     emitted : buf;
     main : buf;
     in_size : int;
-  } [@@deriving sexp,yojson]
+  } [@@deriving sexp,show,yojson]
 
-  type entries = Table.pre_entry list
+  type entries = (Ast.qualified_name * (int option * Ast.match_ list * Ast.action * Ast.id option) list) list
```

Comment:
I needed it in order to pass along information for adding table entries from the control plane since I needed information from the evaluation environment to create the Table.pre_entry. I suppose an alternative approach would be to create analogous definitions for the `Ast` types within `Prog`. I thought this way would make things simpler/have less duplicate code. Thoughts?

Reply:
I think it's fine for now. I don't like having dependencies from lib/ into "testing code". We could put the definitions of these types into prog so the dependency goes the other way, that would  be better.